src/compiler/types/tests/generate-app-types.spec.ts

```diff
@@ -1406,6 +1409,198 @@ declare module "@stencil/core" {
                 }
         }
 }
+`,
+      {
+        immediateWrite: true,
+      }
+    );
+  });
+
+  it('should transform aliased paths if the config option is true', async () => {
+    const compilerComponentMeta = stubComponentCompilerMeta({
+      tagName: 'my-component',
+      componentClassName: 'MyComponent',
+      jsFilePath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.js'),
+      sourceFilePath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.tsx'),
+      sourceMapPath: path.join(config.rootDir, 'some/stubbed/path/a/my-component.js.map'),
+      hasProp: true,
+      properties: [
+        stubComponentCompilerProperty({
+          name: 'name',
+          complexType: {
+            original: 'UserImplementedPropType',
+            resolved: '"foo" | "bar"',
+            references: {
+              UserImplementedPropType: {
+                location: 'import',
+                path: '@utils',
+              },
+            },
+          },
+        }),
+      ],
+    });
+    buildCtx.components = [compilerComponentMeta];
+    config.tsCompilerOptions = {
+      paths: {
+        '@utils': [path.join(config.rootDir, 'some/stubbed/path/utils/utils.ts')],
+      },
+      declaration: true,
+    };
+    config.transformAliasedImportPaths = true;
+    // We need to have a file in the in-memory fs for the TS module resolution to succeed
+    originalWriteFile(path.join(config.rootDir, 'some/stubbed/path/utils/utils.ts'), '');
```

Comment:
Technically, this method is async in TS' eyes - can we add `await` to it?
```suggestion
    await originalWriteFile(path.join(config.rootDir, 'some/stubbed/path/utils/utils.ts'), '');
```